File: 0092/0092.py
from typing import *

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        dummy = leftEdge = ListNode(next=head)
        for _ in range(left-1):
            leftEdge = leftEdge.next

        prevNode, currentNode = None, leftEdge.next

        for _ in range(right - left + 1):
            nextNode = currentNode.next
            currentNode.next = prevNode
            prevNode = currentNode
            currentNode = nextNode

        leftEdge.next.next = currentNode
        leftEdge.next = prevNode

        return dummy.next

File: 0092/0092.c
// Definition for singly-linked list.
struct ListNode {
     int val;
     struct ListNode *next;
};


struct ListNode* reverseBetween(struct ListNode* head, int left, int right){
    struct ListNode* dummy = malloc(sizeof(struct ListNode));
    struct ListNode* leftEdge = dummy;

    dummy->next = head;
    leftEdge->next = head;

    for(int i = 0; i < left-1; i++){
        leftEdge = leftEdge->next;
    }

    struct ListNode* prevNode = malloc(sizeof(struct ListNode));
    struct ListNode* currentNode = leftEdge->next;

    for (int i = 0; i < right - left + 1; i++){
        struct ListNode* nextNode = currentNode->next;
        currentNode->next = prevNode;
        prevNode = currentNode;
        currentNode = nextNode;
    }

    leftEdge->next->next = currentNode;
    leftEdge->next = prevNode;

    return dummy->next;
}